lbz/authz.py

```diff
@@ -115,9 +118,17 @@ def _check_allow_and_set_resources(self):
                 if self._allow_if_allow_all(d_domain):
                     return
                 elif resource_to_check := d_domain.get(self.permission):
+                    effective_permissions = resource_to_check
+                    if ref_name := resource_to_check.get("ref"):
+                        if ref_name not in self.refs:
```

Comment:
3 nested `ifs` in `try/catch` - could you please try to flatten this? If you don't have time or feel like it's not crucial then leave it but imho this will make debugging kinda hard.

Reply:
Moved to a separate function